mantidproject/mantid
# Muon AlC interface crashes

**Description of work.**
The ALCLatestFileFinder expected the instrument name to be case-consistent, e.g EMU, or emu. If it was supplied with a file with an upper case detector name, e.g EMU00066666 from //emu/data it would crash. The crash was due to querying an empty vector, which now has a return check, on the off chance the underlying directory changes (e.g wiped). The instrument check has also been changed so that it is now case-insensitive.

**To test:**
To replicate the original issue

    Set the default instrument to EMU, and turn off the data archive
    Add the //emu/data directory to user directories. 
    Interfaces->Muon->ALC
    Enter 66666
    Tick auto
    Press Load
    Should no longer crash (although this operation takes a long time as it loads a large amount of files in //emu/data )

Fixes #27221 . 
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
